doc/source/flexibility/compressed_columns.rst

```diff
@@ -0,0 +1,213 @@
+``COLUMN_FORMAT`` modifier: ``COLUMN_FORMAT COMPRESSED``. If this modifier is
+applied on an unsupported column type or storage engine, an error is returned.
+
+The compression can be specified:
+
+* at the table create time:
+  ``CREATE TABLE ... (..., foo BLOB COLUMN_FORMAT COMPRESSED, ...);``
+
+* or a table can be modified to compress/decompress a column later:
+  ``ALTER TABLE ... MODIFY [COLUMN] ... COLUMN_FORMAT COMPRESSED``, or
+  ``ALTER TABLE ... CHANGE [COLUMN] ... COLUMN_FORMAT COMPRESSED``.
+
+To decompress a column, a ``COLUMN_FORMAT`` value any other than ``COMPRESSED``
+can be specified: ``FIXED``, ``DYNAMIC``, or ``DEFAULT``. If there is a column
+compression/decompression request in an ``ALTER TABLE``, it is forced to the
+``COPY`` algorithm.
+
+Two new variables: :variable:`innodb_compressed_columns_zip_level` and
+:variable:`innodb_compressed_columns_threshold` have been implemented.
+
+Compression dictionary support
+==============================
+
+To achieve better compression ration on relatively small individual data items,
+it is possible to pre-define compression dictionary, which is a set of strings
+for each compressed column.
+
+The compression dictionary is stored in a new system |InnoDB| table or tables.
+As this table or tables are of the data dictionary kind, concurrent reads are
+allowed, but writes are serialized, and reads are blocked by writes. Table read
+through old read views are unsupported, similarly to |InnoDB| internal DDL
+transactions. The enforcement of these constraints depends on the
+implementation option below.
+
+Compression dictionaries can be represented as a list of words in a form of a
+string (comma or any other character can be used as a delimiter although not
+required). In other words, ``a,bb,ccc``, ``a bb ccc`` and ``abbccc`` will have
+the same effect. However, the latter is more space-efficient. Quote symbol
+quoting is handled by regular SQL quoting. Dictionaries of at least 128KB
+length are supported.
+
+INFORMATION_SCHEMA Tables
+=========================
+
+This feature implemented two new ``INFORMATION_SCHEMA`` tables.
+
+.. table:: INFORMATION_SCHEMA.XTRADB_ZIP_DICT
+
+  :column BIGINT(21)_UNSIGNED id: dictionary ID
+  :column VARCHAR(64) name: dictionary name
+  :column BLOB zip_dict: compressed dictionary
+
+This table stores the compression dictionary itself. ``SUPER`` privilege is
+required to query it.
+
+.. table:: INFORMATION_SCHEMA.XTRADB_ZIP_DICT_COLS
+
+  :column BIGINT(21)_UNSIGNED table_id: table ID
+  :column BIGINT(21)_UNSIGNED column_pos: column position
+  :column BIGINT(21)_UNSIGNED dict_id: dictionary ID
+
+This table stores the mapping between the compression dictionary and columns
+using the dictionary.
+
+Limitations
+===========
+
+Compressed columns cannot be used in indices (neither on their own nor as parts
+of composite keys).
+
+.. note::
+
+  ``CREATE TABLE t2 AS SELECT * FROM t1``, where ``t1`` already has compressed
```

Comment:
Sorry, outdated documentation.
`CREATE TABLE t2 AS SELECT * FROM t1` will create `t2` with compressed columns and dictionaries.
Recommend to remove this paragraph or alter it to show that we will try to preserve compressed attributes where possible.
For instance, `CREATE TABLE t2 AS SELECT * FROM t1` will create a new table with a compressed column, whereas `CREATE TABLE t2 AS SELECT CONCAT(a,'') AS a FROM t1` will drop this attribute,